components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt

```diff
@@ -51,11 +51,19 @@ class AppLinksUseCases(
     private val launchInApp: () -> Boolean = { false },
     private val alwaysDeniedSchemes: Set<String> = ALWAYS_DENY_SCHEMES
 ) {
-    @Suppress("QueryPermissionsNeeded") // We expect our browsers to have the QUERY_ALL_PACKAGES permission
+    @Suppress(
+        "QueryPermissionsNeeded", // We expect our browsers to have the QUERY_ALL_PACKAGES permission
+        "TooGenericExceptionCaught",
+    )
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun findActivities(intent: Intent): List<ResolveInfo> {
-        return context.packageManager
-            .queryIntentActivities(intent, PackageManager.GET_RESOLVED_FILTER)
+        return try {
+            context.packageManager
+                .queryIntentActivities(intent, PackageManager.GET_RESOLVED_FILTER)
+        } catch (e: RuntimeException) {
```

Comment:
I'm not sure about this.  The [documentation](https://developer.android.com/reference/android/content/pm/PackageManager#queryIntentActivities(android.content.Intent,%20android.content.pm.PackageManager.ResolveInfoFlags)) doesn't show that it can throw an error.  Do we have any crash reporting that tells us what error it could be throwing?

Reply:
I see the original issue.  Maybe we just catch the `TransactionTooLargeException` exception?

Reply:
Based on the report from https://github.com/mozilla-mobile/fenix/issues/27042 it seems to me like there is a `RuntimeException` wrapping the `TransactionTooLargeException`?